packages/ingest/src/granule.ts

```diff
@@ -410,11 +415,81 @@ export async function moveGranuleFiles(
       }
     }
   );
-
   await Promise.all(moveFileRequests);
-
   return movedGranuleFiles;
 }
+/**
+* Moves a granule file and updates the postgres database accordingly
+* @summary Moves a granule file record according to MoveFileParams and updates database accordingly
+* @param {MoveFileParams} moveFileParam - Parameter object describing the move operation
+* @param {FilePgModel} filesPgModel - FilePgModel instance
+* @param {Knex.Transaction | Knex} trx - Knex transaction or (optionally) Knex object
+* @param {number | undefined } postgresCumulusGranuleId - postgres internal granule id
+* @param {boolean} writeToPostgres - explicit flag to enable/disable postgres database updates
+* @returns {ReturnValueDataTypeHere} Brief description of the returning value here.
+*/
+export async function moveGranuleFile(
+  moveFileParam: MoveFileParams,
+  filesPgModel: FilePgModel,
+  trx: Knex.Transaction | Knex,
+  postgresCumulusGranuleId: number | undefined,
+  writeToPostgres: boolean = true
+): Promise<MovedGranuleFile> {
+  const { source, target, file } = moveFileParam;
+  if (moveFileParam.source && moveFileParam.target) {
+    log.debug('moveGranuleS3Object', source, target);
+    if (writeToPostgres) {
+      if (!postgresCumulusGranuleId) {
+        throw new Error('postgresCumulusGranuleId must be defined to move granule file if writeToPostgres is true');
+      }
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      const { cumulus_id } = await filesPgModel.get(trx, {
+        granule_cumulus_id: postgresCumulusGranuleId,
+        bucket: moveFileParam.source.Bucket,
+        key: moveFileParam.source.Key,
+      });
+      await filesPgModel.upsertById(trx, {
+        cumulus_id,
+        granule_cumulus_id: postgresCumulusGranuleId,
+        bucket: moveFileParam.target.Bucket,
+        key: moveFileParam.target.Key,
+        file_name: getNameOfFile(file),
+      });
+    }
+    await S3.moveObject({
+      sourceBucket: moveFileParam.source.Bucket,
+      sourceKey: moveFileParam.source.Key,
+      destinationBucket: moveFileParam.target.Bucket,
+      destinationKey: moveFileParam.target.Key,
+      copyTags: true,
+    });
+
+    return {
+      bucket: moveFileParam.target.Bucket,
+      key: moveFileParam.target.Key,
+      name: getNameOfFile(file),
+    };
+  }
+  let fileBucket;
+  let fileKey;
+  if (file.bucket && file.key) {
+    fileBucket = file.bucket;
+    fileKey = file.key;
+  } else if (file.filename) {
+    const parsed = S3.parseS3Uri(file.filename);
+    fileBucket = parsed.Bucket;
+    fileKey = parsed.Key;
```

Comment:
We have this logic in several places now. FileUtils.js, db/src/translate/file.ts. Would be nice to consolidate but out of scope here

Reply:
👍🏻  this was code moved verbatim from the original move granules logic.